substrate/frame/child-bounties/src/migration.rs

```diff
@@ -0,0 +1,155 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::*;
+use core::marker::PhantomData;
+use frame_support::{
+	storage_alias,
+	traits::{Get, UncheckedOnRuntimeUpgrade},
+};
+
+#[cfg(feature = "try-runtime")]
+use alloc::vec::Vec;
+#[cfg(feature = "try-runtime")]
+use frame_support::ensure;
+
+pub mod v1 {
+	use super::*;
+
+	pub struct MigrateToV1Impl<T>(PhantomData<T>);
+
+	#[storage_alias]
+	type ChildBountyDescriptions<T: Config + pallet_bounties::Config> = StorageMap<
+		Pallet<T>,
+		Twox64Concat,
+		BountyIndex,
+		BoundedVec<u8, <T as pallet_bounties::Config>::MaximumReasonLength>,
+	>;
+
+	impl<T: Config> UncheckedOnRuntimeUpgrade for MigrateToV1Impl<T> {
+		fn on_runtime_upgrade() -> frame_support::weights::Weight {
+			// increment reads/writes after the action
+			let mut reads = 0u64;
+			let mut writes = 0u64;
+
+			let mut old_bounty_ids = Vec::new();
+			// first iteration collect all existing ids not to mutate map as we iterate it
+			for (parent_bounty_id, old_child_bounty_id) in ChildBounties::<T>::iter_keys() {
+				reads += 1;
+				old_bounty_ids.push((parent_bounty_id, old_child_bounty_id));
+			}
+
+			// keep ids order roughly the same with the old order
+			old_bounty_ids.sort();
```

Comment:
this generally not good, the runtime logic complexity should not be worth than linear. you can keep your collection sorted while adding elements

Reply:
Ok I can use BTreeSet instead of vector then